lib/gooddata/bricks/middleware/execution_result_middleware.rb

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2019, GoodData Corporation. All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+require_relative 'base_middleware'
+
+module GoodData
+  module Bricks
+    module ExecutionStatus
+      OK = 'OK'
+      ERROR = 'ERROR'
+      WARNING = 'WARNING'
+    end
+
+    class ExecutionResultMiddleware < Bricks::Middleware
+      @@result_log_path = nil
+
+      def call(params)
+        result_log_path(params)
+        @app.call(params)
+      end
+
+      # Update process execution result when the script is executed inside a GDC ETL process execution (ruby bricks).
+      # Ruby bricks should update execution result at the end of script with status=WARNING or OK and summary message if the script exits normally.
+      # If any fatal error, script should update execution result with status=ERROR and error message,
+      # then throw exception to notify GDC platform that the script terminated unexpectedly.
+      # @param [HashMap] params contains GDC_EXECUTION_RESULT_LOG_PATH or GDC_LOG_DIRECTORY, GDC_EXECUTION_ID
+      # @param [ExecutionStatus] status execution status
+      # @param [String] message execution message
+      def self.update_execution_result(status = ExecutionStatus::OK, message = "")
```

Comment:
Please remove status default value, we should call update_execution_result(ExecutionStatus::OK) than only update_execution_result.